lib/modules/manager/gradle/parser.spec.ts

```diff
@@ -351,6 +351,27 @@ describe('modules/manager/gradle/parser', () => {
       });
     });
 
+    describe('concatenated dependency strings', () => {
+      test.each`
+        def                                  | str                               | output
+        ${''}                                | ${'"foo:bar" + ":1.2.3"'}         | ${{ depName: 'foo:bar', currentValue: '1.2.3', managerData: { fileReplacePosition: 15 } }}
+        ${''}                                | ${'"foo:bar:" + "1.2.3"'}         | ${{ depName: 'foo:bar', currentValue: '1.2.3', managerData: { fileReplacePosition: 15 } }}
+        ${''}                                | ${'"foo:bar:" + "1.2.3@zip"'}     | ${{ depName: 'foo:bar', currentValue: '1.2.3', dataType: 'zip', managerData: { fileReplacePosition: 15 } }}
+        ${''}                                | ${'"foo:" + "bar:1.2.3"'}         | ${{ depName: 'foo:bar', currentValue: '1.2.3', managerData: { fileReplacePosition: 15 } }}
+        ${''}                                | ${'"foo:bar:1." + "2.3"'}         | ${{ depName: 'foo:bar', currentValue: '1.2.3', skipReason: 'contains-variable' }}
+        ${''}                                | ${'"foo:bar1.2.3:" + "1.2.3"'}    | ${{ depName: 'foo:bar1.2.3', currentValue: '1.2.3', managerData: { fileReplacePosition: 20 } }}
+        ${'baz = "1.2.3"'}                   | ${'"foo:bar:" + baz'}             | ${{ depName: 'foo:bar', currentValue: '1.2.3', managerData: { fileReplacePosition: 7 } }}
+        ${'baz = "1.2.3"'}                   | ${'"foo:bar:" + property("baz")'} | ${{ depName: 'foo:bar', currentValue: '1.2.3', managerData: { fileReplacePosition: 7 } }}
+        ${'baz = "1.2.3"'}                   | ${'"foo:bar:" + baz + "456"'}     | ${{ depName: 'foo:bar', currentValue: '1.2.3456', skipReason: 'contains-variable' }}
+        ${'baz = "1.2.3"'}                   | ${'"foo:bar:" + baz + "@zip"'}    | ${{ depName: 'foo:bar', currentValue: '1.2.3', dataType: 'zip', managerData: { fileReplacePosition: 7 } }}
+        ${'foo.bar = "bar:"; baz = "1.2.3"'} | ${'"foo:" + foo.bar + "${baz}"'}  | ${{ depName: 'foo:bar', currentValue: '1.2.3', managerData: { fileReplacePosition: 25 } }}
+        ${'foo.bar = "bar"; baz = "1.2.3"'}  | ${'"foo:bar_${foo.bar}:" + baz'}  | ${{ depName: 'foo:bar_bar', currentValue: '1.2.3', managerData: { fileReplacePosition: 24 } }}
+      `('$def | $str', ({ def, str, output }) => {
+        const { deps } = parseGradle([def, str].join('\n'));
+        expect(deps).toMatchObject([output].filter(Boolean));
```

Comment:
please use `is.truthy` instead . why is that filter needed?

Reply:
Good catch, not needed at all here. I've changed the other occurrences to use `is.truthy`